pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -53,82 +54,109 @@ func computeDesiredFleetSize(fas *autoscalingv1.FleetAutoscaler, f *agonesv1.Fle
 }
 
 func applyWebhookPolicy(w *autoscalingv1.WebhookPolicy, f *agonesv1.Fleet) (int32, bool, error) {
-	faReq := autoscalingv1.FleetAutoscaleReview{
-		Request: &autoscalingv1.FleetAutoscaleRequest{
-			UID:       uuid.NewUUID(),
-			Name:      f.Name,
-			Namespace: f.Namespace,
-			Status:    f.Status,
-		},
-		Response: nil,
+	if w == nil {
+		return 0, false, errors.New("nil WebhookPolicy passed")
 	}
-	b, err := json.Marshal(faReq)
-	urlStr := ""
-	if w.URL != nil {
-		urlStr = *w.URL
+
+	if f == nil {
+		return 0, false, errors.New("nil Fleet passed")
 	}
-	var faResp autoscalingv1.FleetAutoscaleReview
-	servicePath := ""
-	if w.Service != nil {
-		if w.Service.Path != nil {
-			servicePath = *w.Service.Path
+
+	if w.URL != nil && w.Service != nil {
```

Comment:
👍 additional check would be great. Validated on Update but still something wrong could happen.

Reply:
Agree, taking into account the fact that another validation happens in a different package.